nixos/modules/services/audio/jack.nix

```diff
@@ -0,0 +1,275 @@
+          '';
+          description = ''
+            Commands to run after JACK is started.
+          '';
+        };
+
+      };
+
+      alsa = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            Route audio to/from generic ALSA-using applications using ALSA JACK PCM plugin.
+          '';
+        };
+
+        support32Bit = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to include the 32-bit alsa plugins libraries in the system or not.
+            This is only useful on 64-bit systems and currently limited to x86_64-linux.
+          '';
+        };
+      };
+
+      loopback = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Create ALSA loopback device, instead of using PCM plugin. Has broader
+            application support (things like Steam will work), but may need fine-tuning
+            for concrete hardware. Requires reboot.
+          '';
+        };
+
+        index = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+            Index of an ALSA loopback device.
+          '';
+        };
+
+        config = mkOption {
+          type = types.string;
+          description = ''
+            ALSA config for loopback device.
+          '';
+          # https://alsa.opensrc.org/Jack_and_Loopback_device_as_Alsa-to-Jack_bridge#id06
+          default = ''
+            pcm.loophw00 {
+              type hw
+              card ${toString cfg.loopback.index}
+              device 0
+              subdevice 0
+            }
+            pcm.amix {
+              type dmix
+              ipc_key 219345
+              slave {
+                pcm loophw00
+              }
+            }
+            pcm.asoftvol {
+              type softvol
+              slave.pcm "amix"
+              control { name Master }
+            }
+            pcm.cloop {
+              type hw
+              card ${toString cfg.loopback.index}
+              device 1
+              subdevice 0
+              format S32_LE
+            }
+            pcm.loophw01 {
+              type hw
+              card ${toString cfg.loopback.index}
+              device 0
+              subdevice 1
+            }
+            pcm.ploop {
+              type hw
+              card ${toString cfg.loopback.index}
+              device 1
+              subdevice 1
+              format S32_LE
+            }
+            pcm.aduplex {
+              type asym
+              playback.pcm "asoftvol"
+              capture.pcm "loophw01"
+            }
+            pcm.!default {
+              type plug
+              slave.pcm aduplex
+            }
+          '';
+        };
+
+        session = mkOption {
+          type = types.string;
+          default = ''
+            alsa_in -j cloop -dcloop &
```

Comment:
Same thing, should be `types.lines` and be in the config section to not lose these defaults